src/disnake/ext/plugins/plugin.py

```diff
@@ -314,11 +320,6 @@ def name(self) -> str:
         """The name of this plugin."""
         return self.metadata.name
 
-    @property
-    def category(self) -> t.Optional[str]:
-        """The category this plugin belongs to."""
-        return self.metadata.category
-
```

Comment:
(This is not necessarily a change I **want** to see changed, I'm mostly just giving an idea-- what do you think?)

Maybe it's an idea to leave this property in for backward compatibility and instead just have it `return self.extras.get("category")`?

Reply:
Maybe it's best then to also turn PluginMetadata.category into a property and emit a DeprecationWarning there, also we might retain all the category params for now but set category to extras instead of metadata and emit a DeprecationWarning

Reply:
> also we might retain all the category params for now but set category to extras instead of metadata and emit a DeprecationWarning

Yeah, that's pretty much what I was thinking. Existing code using a category kwarg would keep working as it'd be picked up and stored in extras anyways, so updating the property to get it from extras and optionally adding a deprecation warning should do the trick.

Reply:
Main changes in 0b1b65b42fd7f2e6c2eb364681c1e7c02c80224f, and also documented PluginMetadata in 88bd109b7ca3804150c56c5ffc1881a26ecb25f7. Since there's quite a lot of unusual changes, i tried explaining them in the updated original PR message.